app/scripts/migrations/121.1.ts

```diff
@@ -0,0 +1,66 @@
+import { hasProperty, isObject } from '@metamask/utils';
+import { cloneDeep } from 'lodash';
+import { CHAIN_IDS } from '../../../shared/constants/network';
+
+export const version = 121.1;
+
+type VersionedData = {
+  meta: { version: number };
+  data: Record<string, unknown>;
+};
+
+/**
+ * Migrates MATIC ticker in Network Configuration to POL ticker as per the direction in https://polygon.technology/blog/save-the-date-matic-pol-migration-coming-september-4th-everything-you-need-to-know
+ *
+ * @param originalVersionedData - Versioned MetaMask extension state, exactly what we persist to dist.
+ * @param originalVersionedData.meta - State metadata.
+ * @param originalVersionedData.meta.version - The current state version.
+ * @param originalVersionedData.data - The persisted MetaMask state, keyed by controller.
+ * @returns Updated versioned MetaMask extension state.
+ */
+export async function migrate(
+  originalVersionedData: VersionedData,
+): Promise<VersionedData> {
+  const versionedData = cloneDeep(originalVersionedData);
+  versionedData.meta.version = version;
+  transformState(versionedData.data);
+  return versionedData;
+}
+
+function transformState(state: Record<string, unknown>) {
```

Comment:
Nit: Two suggestions:

* Ideally this function would either mutate the inputs, or return the transformed state without mutating the inputs, not do both. Currently the return value is not used, so maybe we're better off returning nothing here so that this is more clear.
* It's generally recommended to explicitly document return types rather than relying on TypeScript inference, as it's easier then for readers to understand what the function does, and it highlights changes to return types that might go unnoticed if we rely solely on type inference (see here for more details: https://github.com/MetaMask/contributor-docs/blob/main/docs/typescript.md#for-functions-and-methods-provide-explicit-return-types)